simpleline/event_loop/main_loop.py

```diff
@@ -92,7 +92,8 @@ def run(self):
         log.debug("Main loop ended. Running callback if set.")
 
         if self._quit_callback:
-            self._quit_callback()
+            cb = self._quit_callback.callback
+            cb(self._quit_callback.args)
```

Comment:
This still can be a list but you need to unpack it inside of the callback. This solution looks to me more transparent for complex code base.